src/client/client.ts

```diff
@@ -41,15 +43,24 @@ export abstract class SuperfaceClientBase extends Events {
 
   public hookContext: HooksContext = {};
 
-  constructor() {
+  constructor(options?: { superJson?: SuperJson | SuperJsonDocument }) {
     super();
+    let passedSuperJson: SuperJson | undefined;
+    if (options?.superJson) {
+      passedSuperJson =
+        options.superJson instanceof SuperJson
+          ? options.superJson
+          : new SuperJson(options.superJson);
+    }
     const superCacheKey = Config.instance().superfacePath;
 
     if (SUPER_CACHE[superCacheKey] === undefined) {
-      SUPER_CACHE[superCacheKey] = SuperJson.loadSync(superCacheKey).unwrap();
+      SUPER_CACHE[superCacheKey] =
+        passedSuperJson ?? SuperJson.loadSync(superCacheKey).unwrap();
```

Comment:
suggestion: this could be extracted to function to just do `SUPER_CACHE[superCacheKey] = getSuperJson(options?.superJson)` to keep the constructor neat.

Reply:
also the caching should probably be done differently for passed parameter :thinking: